src/Microsoft.ML.Data/Depricated/Vector/VBufferMathUtils.cs

```diff
@@ -361,19 +386,21 @@ public static void ScaleInto(in VBuffer<Float> src, Float c, ref VBuffer<Float>
             // equal lengths, but I assume I don't care here.
             if (c == 1)
                 src.CopyTo(ref dst);
-            else if (src.Count == 0 || c == 0)
+            else if (src.GetValues().Length == 0 || c == 0)
             {
                 if (src.Length > 0 && src.IsDense)
                 {
-                    var values = dst.Values;
                     // Due to sparsity preservation from src, dst must be dense, in the same way.
-                    Utils.EnsureSize(ref values, src.Length, src.Length, keepOld: false);
-                    if (values == dst.Values) // We need to clear it.
-                        Array.Clear(values, 0, src.Length);
-                    dst = new VBuffer<Float>(src.Length, values, dst.Indices);
+                    var mutation = VBufferMutationContext.Create(ref dst, src.Length);
+                    if (!mutation.CreatedNewValues) // We need to clear it
```

Comment:
>t [](start = 73, length = 1)

you lost the period.

Reply:
OOPS. Will fix.

---
In reply to: [233669776](https://github.com/dotnet/machinelearning/pull/1580#discussion_r233669776) [](ancestors = 233669776)